tony-core/src/main/java/com/linkedin/tony/tensorflow/TonySession.java

```diff
@@ -177,35 +174,24 @@ public boolean allTasksScheduled() {
     return true;
   }
 
-  /**
-   * Associate an allocationRequestId to a TensorFlow job.
-   * @param jobName the TensorFlow job name
-   * @param allocationRequestId the allocationRequestId which corresponds to an instance of this job
-   */
-  public void addAllocationId(String jobName, long allocationRequestId) {
-    if (jobTypeToAllocationIds.get(jobName) == null) {
-      jobTypeToAllocationIds.put(jobName, new HashSet<>());
-    }
-    jobTypeToAllocationIds.get(jobName).add(allocationRequestId);
-    LOG.info(String.format("Job %s with allocationRequestId %d", jobName, allocationRequestId));
-  }
-
   /**
    * Get a TensorFlow task that hasn't been scheduled.
-   * @param allocationRequestId the allocationRequestId of the allocated container
+   * In the absence of allocationRequestId, we are relying on the fact that each tensorflow job will
```

Comment:
this is a bit too hacky

Reply:
Hadoop 2.7 doesn't have `allocationId` in `ContainerRequest` and `Container`. The only other way I can think of for matching containers to tasks is to do resource matching (match the memory and vcores returned to that requested for a task type). Do you prefer that?

Reply:
I think we used to use resources for matching, but that doesn't work for some cases (can't remember the details). @hungj thoughts?

Reply:
Then I guess we have no choice but to match by priority?